src/install.rs

```diff
@@ -136,3 +150,41 @@ fn chmod_x(file: &PathBuf) {
         error!("Failed to run chmod +x: {}", e)
     };
 }
+
+fn mac_process_zip(archive_path: &PathBuf) {
+    // Using Mac unzip because it keeps the execution flags intact. 
+    let archive_parent = archive_path.parent().unwrap();
+    let _output = Command::new("/usr/bin/unzip")
+                            .arg(archive_path.to_string_lossy().to_string())
+                            .arg("-d")
+                            .arg(archive_parent.to_string_lossy().to_string())
+                            .output()
+                            .expect("Problem in Mac postprocessing: Unzip failed");
+
+    // delete the zip file
+    if let Err(e) = fs::remove_file(archive_path) {
+        error!("Problem in Mac postprocessing: failed to remove archive. {}", e)
+    };
+}
+
+fn mac_process_dmg(archive_path: &PathBuf) {
+    // Mount the disk image file
+    let attach_info = Attach::new(archive_path).attach().expect("Problem in Mac post-processing: mounting of dmg file failed");
```

Comment:
A lot of `expect`s and `unwrap`s in this method. This will crash the entire executing thread until iced (hopefully) catches the panic. Instead, you should make the method return a `Result<PathBuf>` or even just `Result<()>`, then use the `?` operator instead of unwrap / expect.

Reply:
I tried to rewrite it, please have another look. I apologize for the cumbersome process, I know exactly what I would like to do, but I don't know how to do it, because the process how I implement things is by looking for samples and then mimicking them. 
So, if you spot something which could be easily improved, please do so directly.